File: Eduflex/Eduflex/DBMigration/Models/Application.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace DBMigration.Models
{

    public class Application
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; } = string.Empty;

        [BsonElement("studentId")]
        public string StudentId { get; set; } = string.Empty;

        [BsonElement("studentName")]
        public string StudentName { get; set; } = string.Empty;

        [BsonElement("description")]
        public string Description { get; set; } = string.Empty;

        [BsonElement("dateApplied")]
        public DateTime DateApplied { get; set; }

        [BsonElement("status")]
        public string Status { get; set; } = "Pending"; // Pending, Approved, Rejected

        [BsonElement("details")]
        public string Details { get; set; } = string.Empty;

        [BsonElement("applicationType")]
        public string ApplicationType { get; set; } = string.Empty;

        [BsonElement("createdAt")]
        public DateTime CreatedAt { get; set; }

        [BsonElement("updatedAt")]
        public DateTime UpdatedAt { get; set; }
    }
}

File: eduflex-Angular/src/app/models/application.ts
import type { Address, EmergencyContact } from './enrolment';

export type ApplicationStatus = 'Pending' | 'Approved' | 'Rejected' | 'Studying';

// 'Studying' means the application was finalized into an Enrolment (see
// EnrolmentModel.StudentApplicationId); 'Rejected' is the terminal decline.
// Neither counts as "still open" for the Student Details active-applications list.
export const ACTIVE_APPLICATION_STATUSES: ApplicationStatus[] = ['Pending', 'Approved'];

export interface Application {
  id: string;
  studentId: string;
  studentName: string;
  studentEmail: string;
  description: string;
  details: string;
  applicationType: string;
  dateApplied: string;
  status: ApplicationStatus;
  createdAt: string;
  updatedAt: string;
}

// Staff-facing read view of a single application — used by the "View application" link
// on the Student Details page and the Enrolment history panel's "From application" link.
export interface ApplicationDetail {
  id: string;
  studentId: string;
  studentName: string;
  description: string;
  dateApplied: string;
  status: ApplicationStatus;
  details: string;
  applicationType: string;
  studyMode?: string;
  campus?: string;
  hometownAddress?: Address;
  currentAddress?: Address;
  emergencyContact?: EmergencyContact;
}
